### PR TITLE
Re-enabling translations in VIUR3

### PR DESCRIPTION
**Pull request by phorward**
Maybe this is still a work-in-progress, but IMHO it improves many things not working currently.

- Set currentLanguage based on `conf["viur.defaultLanguage"]`
- Use static translations from the `language` module again
- Translate `SelectBone` values
- Improved `translate`-class to fallback to original key in case no translation is found
- Fixed imports to be more clear
- Fixed translation of exception messages (which works!)

image

### CODE REVIEW

**Review by sveneberth — CHANGES_REQUESTED** [review]
Hey @phorward, in General im fine with your changes. But I don't get the reason behind the changed imports. I think it's a kind of personal preference, but I like the shorter `from xy import name` rather than `import xy` with the dot-notation in usage (except it's very vulnerable for conflicts). Anyway, we should follow a consistent style. Maybe it's time to create a ViUR style guide. Otherwise the next person change the imports back.

File: core/i18n.py [hunk 1]
```diff
@@ -1,10 +1,7 @@
-from datetime import datetime
-from jinja2.ext import Extension, nodes
+import datetime
+import jinja2.ext as jinja2
```
[the remark this answers is not in the document]
Comment (phorward) [reply core/i18n.py:2]: I was unable to just `import jinja2` and then use `jinja2.ext.Extension` and `jinja2.ext.nodes`, therefore this way.
Its IMHO better than just provide `Extension` and `nodes` on global level.

**Review by XeoN-GHMB — APPROVED** [review]

**Comment by XeoN-GHMB** [comment]
The core will be reformatted soon, so we should create a viur style guide.

File: core/i18n.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (phorward) [reply core/i18n.py:2]: > The core will be reformatted soon, so we should create a viur style guide.

https://github.com/viur-framework/viur-core/pull/491#issuecomment-1236618894

**Pull request merged**